projects/uk.ac.man.cs.llvm/src/uk/ac/man/cs/llvm/ir/types/StructureType.java

```diff
@@ -58,6 +58,23 @@ public Type getElementType(int index) {
         return types[index];
     }
 
+    public long getElementOffset(int index) {
+        int offset = 0;
+        for (int i = 0; i <= index; i++) {
+            if (offset % types[i].getAlignment() != 0) {
+                offset += types[i].getAlignment() - (offset % types[i].getAlignment());
+            }
+
+            if (i == index) {
+                break;
+            }
+
+            offset += types[i].sizeof();
+        }
+
+        return offset * Byte.SIZE;
```

Comment:
If the index is not found you return the size of the struct? If yes, you should also add the padding based on the largest alignment of the struct's members if the struct is not packed.

However, would it not be better to throw an `AssertionError`?